frontend/src/pages/distributedWorkloads/global/useDistributedWorkloadsTabs.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { SupportedArea, useIsAreaAvailable } from '~/concepts/areas';
+import { useUser } from '~/redux/selectors';
+import GlobalDistributedWorkloadsClusterMetricsTab from './GlobalDistributedWorkloadsClusterMetricsTab';
+import GlobalDistributedWorkloadsProjectMetricsTab from './GlobalDistributedWorkloadsProjectMetricsTab';
+import GlobalDistributedWorkloadsJobMetricsTab from './GlobalDistributedWorkloadsJobMetricsTab';
+
+export enum DistributedWorkloadsTabId {
+  DS_CLUSTER_METRICS = 'ds-cluster-metrics',
+  PROJECT_METRICS = 'project-metrics',
+  JOB_METRICS = 'job-metrics',
+}
+
+export type DistributedWorkloadsTabConfig = {
+  id: DistributedWorkloadsTabId;
+  title: string;
+  path: string;
+  isAvailable: boolean;
+  // TODO mturley combine these two booleans into a string union? naming is hard.
+  hasProjectParam?: boolean;
+  allowAllProjects?: boolean;
+  ContentComponent: React.FC<{ tabConfig: DistributedWorkloadsTabConfig }>;
+};
+
+export const useDistributedWorkloadsTabs = (): DistributedWorkloadsTabConfig[] => {
+  const dwAreaIsAvailable = useIsAreaAvailable(SupportedArea.DISTRIBUTED_WORKLOADS).status;
+  const { isAdmin } = useUser();
+  // TODO mturley should we useMemo here?
```

Comment:
imo `useMemo` has only two use-cases
1. Reference stability
2. Performance improve (usually complex calculations only calculated some times)

If you don't need stability in your data, don't worry about it. If you think you need performance improvements, I got news for you, you probably don't lol 🙂 

React will render as many times as it needs, it will be fast and things will be recalculated to no fault or worthwhile slowness, and you just need to avoid causing bad hook dependencies that cause re-renders. Everything else is not worth the time.